### PR TITLE
Added regularization and attempted to address #160

### PR DESCRIPTION
**Pull request by Jegp**
* Migrated to Pytorch Lightning
* Increased model and added batch normalization
* Added a regularizer that wraps other modules, handy in sequential constructions

### CODE REVIEW

File: norse/task/cifar10.py [hunk 1]
```diff
@@ -2,399 +2,212 @@
...
+        if self.optimizer == "adam":
+            optimizer = torch.optim.Adam(
+                self.parameters(), lr=self.lr, weight_decay=1e-5
+            )
+        else:
+            optimizer = torch.optim.SGD(self.parameters(), lr=self.lr, momentum=0.9)
+        self.scheduler = torch.optim.lr_scheduler.StepLR(
+            optimizer, step_size=5, gamma=0.2
+        )
+        return optimizer
```
Comment (Jegp) [inline norse/task/cifar10.py:125]: You can let PyTorch-Lightning handle the Scheduler itself like such:
```python
return [optimizer], [scheduler]
```
And then delete the `training_epoch_end` method.

File: norse/task/cifar10.py [hunk 2]
```diff
@@ -2,399 +2,212 @@
...
+import torch
+import torch.nn.functional as F
+import torch.utils.data
+import pytorch_lightning as pl
+
+import norse
+
+
+def label_smoothing_loss(y_hat, y, alpha=0.2):
+    log_probs = torch.nn.functional.log_softmax(y_hat, dim=1, _stacklevel=5)
```
Comment (Jegp) [inline norse/task/cifar10.py:19]: Is `_stacklevel` not deprecated?

**Review by Jegp — COMMENTED** [review]
Thanks for the work Christian. I'm happy when the checks passes!

**Review by cpehle — APPROVED** [review]
LGTM.
